tests/rptest/tests/cloud_storage_toggle_test.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from rptest.services.kgo_repeater_service import repeater_traffic
+from rptest.services.kgo_verifier_services import KgoVerifierSeqConsumer
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.types import TopicSpec
+from rptest.clients.rpk import RpkTool
+from rptest.services.redpanda import CloudStorageType, SISettings, MetricsEndpoint, CloudStorageType, CHAOS_LOG_ALLOW_LIST
+from rptest.utils.mode_checks import skip_debug_mode
+from rptest.utils.si_utils import S3Snapshot
+
+
+class CloudStorageToggleTest(RedpandaTest):
+    partition_count = 16
+    log_segment_size = 1024 * 1024
+
+    topics = (TopicSpec(
+        partition_count=partition_count,
+        replication_factor=3,
+    ), )
+
+    def __init__(self, test_context, *args, **kwargs):
+        self.si_settings = SISettings(test_context,
+                                      log_segment_size=self.log_segment_size)
+        super().__init__(test_context, si_settings=self.si_settings)
+
+    @cluster(num_nodes=5)
+    @skip_debug_mode
```

Comment:
Maybe worth adding a basic test to check that once toggled off, we actually stop hydrating segments and begin reading from the beginning of the local log